src/auto/tls_database.rs

```diff
@@ -0,0 +1,166 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ ea993ed)
+// from gir-files (https://github.com/gtk-rs/gir-files @ 469db10)
+// DO NOT EDIT
+
+use Cancellable;
+use Error;
+use SocketConnectable;
+use TlsCertificate;
+use TlsCertificateFlags;
+use TlsDatabaseLookupFlags;
+use TlsDatabaseVerifyFlags;
+use TlsInteraction;
+use ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct TlsDatabase(Object<ffi::GTlsDatabase, ffi::GTlsDatabaseClass>);
+
+    match fn {
+        get_type => || ffi::g_tls_database_get_type(),
+    }
+}
+
+pub trait TlsDatabaseExt {
+    fn create_certificate_handle(&self, certificate: &TlsCertificate) -> Option<String>;
+
+    fn lookup_certificate_for_handle<'a, 'b, P: Into<Option<&'a TlsInteraction>>, Q: Into<Option<&'b Cancellable>>>(&self, handle: &str, interaction: P, flags: TlsDatabaseLookupFlags, cancellable: Q) -> Result<Option<TlsCertificate>, Error>;
+
+    fn lookup_certificate_for_handle_async<'a, 'b, P: Into<Option<&'a TlsInteraction>>, Q: Into<Option<&'b Cancellable>>, R: FnOnce(Result<(), Error>) + Send + 'static>(&self, handle: &str, interaction: P, flags: TlsDatabaseLookupFlags, cancellable: Q, callback: R);
+
+    fn lookup_certificate_issuer<'a, 'b, P: Into<Option<&'a TlsInteraction>>, Q: Into<Option<&'b Cancellable>>>(&self, certificate: &TlsCertificate, interaction: P, flags: TlsDatabaseLookupFlags, cancellable: Q) -> Result<TlsCertificate, Error>;
+
+    fn lookup_certificate_issuer_async<'a, 'b, P: Into<Option<&'a TlsInteraction>>, Q: Into<Option<&'b Cancellable>>, R: FnOnce(Result<(), Error>) + Send + 'static>(&self, certificate: &TlsCertificate, interaction: P, flags: TlsDatabaseLookupFlags, cancellable: Q, callback: R);
+
+    //fn lookup_certificates_issued_by<'a, 'b, P: Into<Option<&'a TlsInteraction>>, Q: Into<Option<&'b Cancellable>>>(&self, issuer_raw_dn: /*Ignored*/&glib::ByteArray, interaction: P, flags: TlsDatabaseLookupFlags, cancellable: Q) -> Result<Vec<TlsCertificate>, Error>;
```

Comment:
Hm why is this not using a GBytes? Should probably add bindings for GByteArray at some point :)

Reply:
I guess :)